src/java.base/share/classes/jdk/internal/loader/BuiltinClassLoader.java

```diff
@@ -281,31 +275,30 @@ public URL findResource(String mn, String name) throws IOException {
             url = findResourceOnClassPath(name);
         }
 
-        return checkURL(url);  // check access before returning
+        return url;
     }
 
     /**
      * Returns an input stream to a resource of the given name in a module
      * defined to this class loader.
      */
-    @SuppressWarnings("removal")
     public InputStream findResourceAsStream(String mn, String name)
         throws IOException
     {
-        // Need URL to resource when running with a security manager so that
-        // the right permission check is done.
-        if (System.getSecurityManager() != null || mn == null) {
-            URL url = findResource(mn, name);
-            return (url != null) ? url.openStream() : null;
-        }
-
-        // find in module defined to this loader, no security manager
-        ModuleReference mref = nameToModule.get(mn);
-        if (mref != null) {
-            return moduleReaderFor(mref).open(name).orElse(null);
+        InputStream in = null;
+        if (mn != null) {
+            // find in module defined to this loader
+            ModuleReference mref = nameToModule.get(mn);
+            if (mref != null) {
+                in = moduleReaderFor(mref).open(name).orElse(null);
+            }
         } else {
-            return null;
+            URL url = findResource(name);
```

Comment:
Hello Alan, before the change in this PR, when the module name `mn` was `null`, we used to call `findResource(mn, name)` (on previously line number 298). The `findResource(String, String)` would then effectively end up calling `findResourceOnClassPath(name)` without doing anything more. 

With this change, we now call `findResource(String)` when module name is `null`. The implementation of `findResource(String)` has additional code which does resource's package name to module mapping and few other things before it can decide whether `findResourceOnClassPath(String)` needs to be called for this resource name. 

Should the change on this line instead have been `findResource(null, name)`?

Reply:
The change was deliberate but you are right that when we know we are looking for a resource on the class path that it would be better to stick with findResource(null, name).